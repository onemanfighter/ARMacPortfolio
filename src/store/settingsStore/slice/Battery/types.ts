export type BatteryState = {};

export interface BatterySettingsAction {}

export type BatteryStateSlice = BatteryState & BatterySettingsAction;
