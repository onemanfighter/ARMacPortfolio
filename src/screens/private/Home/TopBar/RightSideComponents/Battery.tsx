import { BatteryIcon, MacIcon, SearchIcon } from '@assets';
import {
  Box,
  Menu,
  MenuDivider,
  MenuItem,
  MenuList,
  Text,
} from '@chakra-ui/react';
import {
  MenuItemComponent,
  MenuListComponent,
  TopBarButton,
} from '@components';
import { useTranslation } from 'react-i18next';

const Battery = () => {
  const batteryPercentage = 67;
  const { t } = useTranslation();
  return (
    <Menu>
      <TopBarButton
        text=""
        onClick={() => {}}
        ariaLabel="battery-top-bar-button"
        icon={
          <BatteryIcon
            percentage={batteryPercentage}
            props={{
              width: '2em',
              height: '2em',
              color: 'white',
            }}
          />
        }
      />
      <MenuListComponent>
        <Box
          display={'flex'}
          justifyContent={'space-between'}
          flexDirection={'row'}
          mx={2}
        >
          <Text color={'white'} fontSize={'xs'} fontWeight={'bold'}>
            {t('TopAppBar.battery.title')}
          </Text>
          <Text color={'white'} fontSize={'xs'}>
            {batteryPercentage + '%'}
          </Text>
        </Box>
        <MenuDivider mx={2} />
        <MenuItemComponent
          text={t('TopAppBar.battery.batterySettings')}
          ariaLabel="new-tab"
          onClick={() => {}}
        />
      </MenuListComponent>
    </Menu>
  );
};

export default Battery;
