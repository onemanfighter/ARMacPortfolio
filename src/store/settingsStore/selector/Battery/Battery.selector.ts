import { SettingsStoreState } from '../../settingsStore';

const batterySelector = (state: SettingsStoreState) => ({});

export { batterySelector };
