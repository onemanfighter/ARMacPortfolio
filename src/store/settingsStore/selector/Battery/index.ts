export { batterySelector } from './Battery.selector';
