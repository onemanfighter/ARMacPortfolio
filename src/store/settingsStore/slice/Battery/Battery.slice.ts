import { SettingsStoreSlice } from '../../settingsStore';
import { BatteryState, BatteryStateSlice } from './types';

const defaultBatteryState: BatteryState = {};

const createBatterySlice: SettingsStoreSlice<BatteryStateSlice> = (set) => ({
  ...defaultBatteryState,
});

export default createBatterySlice;
