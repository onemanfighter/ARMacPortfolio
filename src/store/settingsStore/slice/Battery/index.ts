export { default as createBatterySlice } from './Battery.slice';
export type { BatteryStateSlice, BatteryState } from './types';
