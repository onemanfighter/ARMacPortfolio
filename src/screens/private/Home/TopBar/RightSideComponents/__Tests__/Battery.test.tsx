import { fireEvent, render, screen } from '@testing-library/react';
import Battery from '../Battery';

describe('Battery', () => {
  it('should render', () => {
    const { container } = render(<Battery />);

    expect(container).toMatchSnapshot();
  });

  it('should open pop-up when battery icon is tapped', () => {
    const { container } = render(<Battery />);

    fireEvent.click(screen.getByLabelText('battery-top-bar-button'));

    expect(container).toMatchSnapshot();
  });

  it('should open battery settings when battery settings is tapped', () => {
    const { container } = render(<Battery />);

    fireEvent.click(screen.getByLabelText('battery-settings'));

    expect(container).toMatchSnapshot();
  });
});
