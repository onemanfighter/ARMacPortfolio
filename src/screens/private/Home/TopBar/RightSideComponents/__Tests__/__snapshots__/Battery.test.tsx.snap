// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Battery should open pop-up when battery icon is tapped 1`] = `
<div>
  <button
    aria-controls="menu-list-:r4:"
    aria-expanded="true"
    aria-haspopup="menu"
    aria-label="battery-top-bar-button"
    class="chakra-menu__menu-button css-qnx1nw"
    data-active=""
    id="menu-button-:r4:"
  >
    <span
      class="css-xl71ch"
    >
      <svg
        color="white"
        fill="currentColor"
        height="2em"
        viewBox="0 0 16 16"
        width="2em"
      >
        <path
          d="M2 6h6.7v4H2V6z"
        />
        <path
          d="M2 4a2 2 0 00-2 2v4a2 2 0 002 2h10a2 2 0 002-2V6a2 2 0 00-2-2H2zm10 1a1 1 0 011 1v4a1 1 0 01-1 1H2a1 1 0 01-1-1V6a1 1 0 011-1h10zm4 3a1.5 1.5 0 01-1.5 1.5v-3A1.5 1.5 0 0116 8z"
        />
      </svg>
    </span>
  </button>
  <div
    class="css-0"
    data-popper-escaped=""
    data-popper-placement="bottom-start"
    data-popper-reference-hidden=""
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 8px); --popper-transform-origin: top left;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-wkqcnc"
      id="menu-list-:r4:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-1kqq1dh"
      >
        <p
          class="chakra-text css-1jueml0"
        >
          Battery
        </p>
        <p
          class="chakra-text css-1p1588x"
        >
          67%
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-n5p1w9"
      />
      <button
        aria-label="new-tab"
        class="chakra-menu__menuitem css-pk46t1"
        data-index="0"
        id="menu-list-:r4:-menuitem-:r5:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Battery settings...
      </button>
    </div>
  </div>
</div>
`;

exports[`Battery should render 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="battery-top-bar-button"
    class="chakra-menu__menu-button css-qnx1nw"
    id="menu-button-:r1:"
  >
    <span
      class="css-xl71ch"
    >
      <svg
        color="white"
        fill="currentColor"
        height="2em"
        viewBox="0 0 16 16"
        width="2em"
      >
        <path
          d="M2 6h6.7v4H2V6z"
        />
        <path
          d="M2 4a2 2 0 00-2 2v4a2 2 0 002 2h10a2 2 0 002-2V6a2 2 0 00-2-2H2zm10 1a1 1 0 011 1v4a1 1 0 01-1 1H2a1 1 0 01-1-1V6a1 1 0 011-1h10zm4 3a1.5 1.5 0 01-1.5 1.5v-3A1.5 1.5 0 0116 8z"
        />
      </svg>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-wkqcnc"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-1kqq1dh"
      >
        <p
          class="chakra-text css-1jueml0"
        >
          Battery
        </p>
        <p
          class="chakra-text css-1p1588x"
        >
          67%
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-n5p1w9"
      />
      <button
        aria-label="new-tab"
        class="chakra-menu__menuitem css-pk46t1"
        data-index="0"
        id="menu-list-:r1:-menuitem-:r2:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Battery settings...
      </button>
    </div>
  </div>
</div>
`;
